File: 242-valid-anagram/242-valid-anagram.cpp
class Solution 
{
public:
    bool isAnagram(string s, string t) 
    {
        if(s.length() != t.length())
            return false;
        
        int chars[26] = {};
        for(unsigned int i = 0; i < s.length(); i++)
        {
            chars[s[i] - 97]++;
            chars[t[i] - 97]--;
        }
        
        for(int& c : chars)
            if(c != 0)
                return false;
        return true;
    }
};

File: 242-valid-anagram/242-valid-anagram.java
class Solution 
{
    public boolean isAnagram(String s, String t) 
    {
        if(s.length() != t.length())
            return false;
        
        int[] alphabet = new int[26];
        for(int i = 0; i < s.length(); i++)
        {
            alphabet[s.charAt(i) - 97]++;
            alphabet[t.charAt(i) - 97]--;
        }
        
        for(int count : alphabet)
            if(count != 0)
                return false;
        
        return true;
    }
}